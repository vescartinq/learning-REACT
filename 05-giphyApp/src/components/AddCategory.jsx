import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState(''); // ''

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  // Funci√≥n que se ejecuta al clickar enter (<form onSubmit={handleSubmit}></form>)
  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((newInput) => [inputValue, ...newInput]);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
