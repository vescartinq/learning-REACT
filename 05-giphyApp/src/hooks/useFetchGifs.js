// CustomHooks
import { useState, useEffect } from 'react';
import { getGifs } from '../actions/gifActions';

export const useFetchGifs = (category) => {
  // Estado inicial CARGANDO
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifs(category).then((images) => {
      setState({
        data: images,
        loading: false,
      });
    });
  }, [category]);

  return state; // Estado inicial: { data:[], loading: true };
};
