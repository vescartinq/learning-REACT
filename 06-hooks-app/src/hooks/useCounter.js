import { useState } from 'react';

export const useCounter = (initialState = 10) => {
  const [counter, setCounter] = useState(initialState); // 10

  const reset = () => {
    setCounter(initialState);
  };

  const increment = (factor) => {
    setCounter(counter + factor);
  };

  const decrement = (factor) => {
    setCounter(counter - factor);
  };

  return {
    counter,
    increment,
    decrement,
    reset,
  };
};
