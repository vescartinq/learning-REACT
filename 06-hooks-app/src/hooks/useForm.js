import { useState } from 'react';

export const useForm = (initialState = {}) => {
  const [values, setValues] = useState(initialState);

  const handleInputChange = (e) => {
    // console.log(e.target.value);

    setValues({
      ...values,
      [e.target.name]: e.target.value,
    });
  };

  return [values, handleInputChange];
};
