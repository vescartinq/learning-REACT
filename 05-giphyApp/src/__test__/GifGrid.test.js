import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { GifGrid } from './../components/GifGrid';

import { useFetchGifs } from '../hooks/useFetchGifs';
jest.mock('../hooks/useFetchGifs'); //simulamos la llamada a la API y controlamos la informaciÃ³n

describe('GifGrid', () => {
  let wrapper;
  const category = 'Avengers';

  beforeEach(() => {
    jest.clearAllMocks(); //reseteamos los mocks

    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    }); // estado inicial de la llamada a la API

    wrapper = shallow(<GifGrid category={category} />); //reseteamos el componente
  });

  test('should render GifGrid component', () => {
    //devolvemos el valor que queramos como respuesta de la API, en este caso los valores default

    expect(wrapper).toMatchSnapshot();
  });

  test('should show items when upload images with useFetchGifs', () => {
    const gifs = [
      {
        id: 'ABC',
        url: 'https://localhost/algo.jpg',
        title: 'Algo',
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });
});
