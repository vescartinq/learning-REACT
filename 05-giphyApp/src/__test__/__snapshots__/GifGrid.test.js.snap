// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid should render GifGrid component 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Avengers
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`GifGrid should show items when upload images with useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Avengers
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Algo"
      url="https://localhost/algo.jpg"
    />
  </div>
</Fragment>
`;
